src/components/graph/GraphCanvas.vue

```diff
@@ -207,6 +212,9 @@ provide(SelectedNodeIdsKey, selectedNodeIds)
 
 // Provide execution state to all Vue nodes
 useExecutionStateProvider()
+// Provide preview images state to all Vue nodes
+const { nodePreviewImages } = storeToRefs(useNodeOutputStore())
+provide(NodePreviewImagesKey, nodePreviewImages)
```

Comment:
This is another case where it seems like we could just `ref`erence this in place instead of using a `provide`